KamerConnect.DataAccess.Postgres/Repositorys/PersonRepository.cs

```diff
@@ -202,4 +207,121 @@ private static T ValidateEnum<T>(string value) where T : struct
     }
+
+    public void UpdatePerson(Person person)
+    {
+        using (var connection = new NpgsqlConnection(ConnectionString))
+        {
+            connection.Open();
+            using (var command = new NpgsqlCommand(
+                       """
+                       UPDATE person SET
+                           email = @Email,
+                           first_name = @FirstName,
+                           middle_name = @MiddleName,
+                           surname = @Surname,
+                           phone_number = @PhoneNumber,
+                           birth_date = @BirthDate,
+                           gender = @gender::gender,
+                           role = @Role::user_role,
+                           profile_picture_path = @ProfilePicturePath
+                       WHERE id = @Id;
+                       """, connection))
+            {
+                command.Parameters.AddWithValue("@Id", Guid.Parse(person.Id));
+                command.Parameters.AddWithValue("@Email", person.Email);
+                command.Parameters.AddWithValue("@FirstName", person.FirstName);
+                command.Parameters.AddWithValue("@MiddleName", person.MiddleName ?? string.Empty);
+                command.Parameters.AddWithValue("@Surname", person.Surname);
+                command.Parameters.AddWithValue("@PhoneNumber", person.PhoneNumber ?? string.Empty);
+                command.Parameters.AddWithValue("@BirthDate", person.BirthDate);
+                command.Parameters.AddWithValue("@Gender", person.Gender.ToString());
+                command.Parameters.AddWithValue("@Role", person.Role.ToString());
+                command.Parameters.AddWithValue("@ProfilePicturePath", person.ProfilePicturePath ?? string.Empty);
+
+                command.ExecuteNonQuery();
+            }
+        }
+    }
+    
+    public void UpdatePersonality(Person person)
+{
+    using (var connection = new NpgsqlConnection(ConnectionString))
+    {
+        connection.Open();
+        
+        using (var transaction = connection.BeginTransaction())
+        {
+            using (var updateCommand = new NpgsqlCommand(
+                """
+                UPDATE personality
+                SET school = @School,
+                    study = @Study,
+                    description = @Description
+                WHERE person_id = @PersonalityId;
+                """, connection))
+            {
+                updateCommand.Parameters.AddWithValue("@PersonalityId", Guid.Parse(person.Id));
+                updateCommand.Parameters.AddWithValue("@School", person.Personality.School ?? string.Empty);
+                updateCommand.Parameters.AddWithValue("@Study", person.Personality.Study ?? string.Empty);
+                updateCommand.Parameters.AddWithValue("@Description", person.Personality.Description ?? string.Empty);
+
+                var rowsAffected = updateCommand.ExecuteNonQuery();
+
+                if (rowsAffected == 0)
```

Comment:
Is het niet mooier om eerst een check te maken of die er is en op basis daarvan een Create of Update aan te roepen? Die kunnen dan ook in losse functies staan. Als je dit in chat gooit kan die dat denk ik wel goed voor je doen.

Reply:
Dit is een vraag wat er nu staat is opzich prima hoor

Reply:
ik zal nog even kijken of dit lukt